pkg/file/span.go

```diff
@@ -0,0 +1,42 @@
+package file
+
+import (
+	"math"
+
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+var Spans []int64
+
+func init() {
+	Spans = GenerateSpanSizes(9, swarm.Branches)
+}
+
+// generates a dictionary of maximum span lengths per level represented by one SectionSize() of data
+//
+// TODO: level 0 should be SectionSize() not Branches()
+func GenerateSpanSizes(levels, branches int) []int64 {
+	spans := make([]int64, levels)
+	branchesSixtyfour := int64(branches)
+	var span int64 = 1
+	for i := 0; i < 9; i++ {
+		spans[i] = span
+		span *= branchesSixtyfour
+	}
+	return spans
+}
+
+// getLevelsFromLength calculates the last level index which a particular data section count will result in.
+// The returned level will be the level of the root hash.
+func GetLevelsFromLength(length int64, sectionSize, branches int) int {
```

Comment:
i think this function can simply be renamed to `Levels`. the documentation should explain what it gets levels for (length), and I don't think there's anything else we'd need to get levels for, except for length. for you consideration.

also, `length` should be of type `uint64` and not `int64`, since data length cannot be negative

Reply:
>  I don't think there's anything else we'd need to get levels for, except for length. for you consideration.

there may be later on, but we can change it back when and if we get there.

> also, length should be of type uint64 and not int64, since data length cannot be negative

`len()` in `golang` returns `int`, not `uint`. `io.Reader.Read()` returns `int, err`. Why would we interpret length differently here?

Reply:
whoops you're right. in old swarm subtrie size is `int64`. apologies i stand corrected